### PR TITLE
Views

### PR DESCRIPTION
**Pull request by Ugboaja-Uchechi**
###  Project Requirements

- Implement design from the specified interface.

### CODE REVIEW

File: app/views/user/show.html.erb [hunk 1]
```diff
@@ -1,3 +1,10 @@
...
+<%= render partial: "partials/user", locals: { user: @user } %>
+<%= render partial: "partials/bio" %>
+<% @user.three_recent_posts.each do |post| %>
+  <%= render partial: "partials/post", locals:{ post: post } %>
+<% end %>
+<div class="button">
+  <button>
+    <%= link_to 'See all Post', user_post_path(@user) %>
+  </button>
+</div>
```
Comment (Emmyn5600) [inline app/views/user/show.html.erb]: - [x] We need to implement the `comment and likes` features to be displayed in the view using this file.

File: app/views/post/index.html.erb [hunk 2]
```diff
@@ -1,3 +1,7 @@
...
-<p>Find me in app/views/post/index.html.erb</p>
-<p>Watch Euphoria</p>
\ No newline at end of file
+<%= render partial: "partials/user", locals: { user: @user } %>
+<% @posts.each do |post| %>
+  <%= render partial: "partials/post", locals:{ post: post } %>
+  
+  <%# post comments %>
+  <%= render partial: "partials/comment", locals: { items: 5 } %>
+<% end %>
```
Comment (Emmyn5600) [inline app/views/post/index.html.erb]: - [x] Uses `post.recent_comments` to render 5 most recent comments in the `app/views/posts/index.html.erb` page

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi @Ugboaja-Uchechi 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet, please work on it and ask for another review.

 ## Required Changes ♻️
 
 - Check the comment under the review.
  
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

File: app/views/partials/_user.html.erb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<div class="user-flex">
+  <div class="user-border photo-width">
+    <i class="fa-solid fa-user"></i>
+  </div>
+  <div class="user-border user-width">
+    <h1><%= link_to user.name, user_path(user) %></h1>
+    <div class="user-border-p">
+      <p>Number of posts: <%= user.postsCounter? %></p>
```
Comment (ahmetbozaci) [inline app/views/partials/_user.html.erb]: ### OPTIONAL
**_It wasn't pointed out earlier. But it is a requirement so ı highly recommend fixing it._**
- [x]  You should fix the post counter.
image

File: app/views/partials/_bio.html.erb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<div class="bio-flex">
+  <div class="bio">
+    <p>Bio</p>
+    <p>
+      Lorem ipsum dolor sit amet consectetur adipisicing elit. Reiciendis aperiam ullam quia accusantium, beatae animi ea dolores modi architecto culpa exercitationem mollitia, rem maxime illum ex odio, adipisci iusto quibusdam.
+    </p>
+  </div>
+</div>
```
Comment (ahmetbozaci) [inline app/views/partials/_bio.html.erb:8]: ### OPTIONAL
**_It wasn't pointed out earlier. But it is a requirement so ı highly recommend fixing it._**
- [x] The bio should come from the user model. You should fix it.

File: app/views/partials/_user.html.erb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<div class="user-flex">
+  <div class="user-border photo-width">
+    <i class="fa-solid fa-user"></i>
+  </div>
```
Comment (ahmetbozaci) [inline app/views/partials/_user.html.erb]: ### OPTIONAL
**_It wasn't pointed out earlier. But it is a requirement so ı highly recommend fixing it._**
- [x] The user photo should come from the user model. You should fix it.

File: app/assets/stylesheets/application.css [hunk 6]
```diff
@@ -1,15 +1,115 @@
...
+.bio p {
+  font-size: 1.3rem;
+  margin: 1rem 1rem;
+  text-align: justify;
+}
+
+/* Posts */
+
+.single-post {
+  display: flex;
```
Comment (ahmetbozaci) [inline app/assets/stylesheets/application.css:81]: ### OPTIONAL
- [x] For DRY CSS you should combine duplicate properties. 
Example, you can create class `.d-flex { display: flex}` and use this class in HTML file.

- [More information about DRY principles](https://www.freecodecamp.org/news/7-important-tips-for-writing-better-css/) 

- [Html and css best practices in curriculum](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md)

File: app/views/partials/_singlepost.html.erb [hunk 7]
```diff
@@ -0,0 +1,22 @@
+<div>
+  <div class="single-post">
+    <div>
+      <p>Post #<%= post.id %> by username <%= post.author %></p>
```
Comment (ahmetbozaci) [inline app/views/partials/_singlepost.html.erb]: ### OPTIONAL
**_It wasn't pointed out earlier. But it is a requirement so ı highly recommend fixing it._**
- [x] You should fix the post author name. 
image

File: app/views/partials/_singlepost.html.erb [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+      <p>Comments: x, Likes: y</p>
```
Comment (ahmetbozaci) [inline app/views/partials/_singlepost.html.erb]: ### OPTIONAL
**_It wasn't pointed out earlier. But it is a requirement so ı highly recommend fixing it._**
- [x] The comment and like should come from the models. You should fix it.

File: app/views/partials/_comment.html.erb [hunk 9]
```diff
@@ -0,0 +1,9 @@
+<div>
+  <div>
+   <% (1..items).each_with_index do |index| %>
+    <p>
+      Username: Comment <%= index %>
+    </p>
+    <% end %>
+  </div>
```
Comment (ahmetbozaci) [inline app/views/partials/_comment.html.erb]: ### Requirement
 **_Nice to implement the comments on the show page requested by a previous reviewer_** 
- [x] But comments should come from the model. You should fix it.

**Review by ahmetbozaci — CHANGES_REQUESTED** [review]
# STATUS: Changes Required  ♻️

### Hello @Ugboaja-Uchechi  ✋🏼 
#### Great work on making the changes requested by a previous reviewer 👏 
#### You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
___
### ♻️ Request changes ♻️

- [ ] _Check the comments under the review._

___

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
___

### Cheers and Happy coding! 💻 👏

almost

___

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_


_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mentioning me as [ahmetbozaci](https://github.com/ahmetbozaci), I will receive a notification and will respond._
___
[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

**Comment by Ugboaja-Uchechi** [comment]
This is my last review.
I have worked on the required and optional reviews to the best of my knowledge.

**Review by SajjadAhmad14 — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

### Hi Stephanie, 
## Congratulations 🎉 Thank you for implementing all the requirements 🙏
### Please go ahead and merge the branch!

## Happy Coding 👏👏👏
#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**